### ISSUE TITLE
Interceptors are skipped for mocked activities

### ISSUE THREAD

**Issue by PatrikValo** [temporalio/sdk-go#1468]
## Expected Behavior
I would expect that when I test a workflow and I mock an activity with return values like following
```go
s.env.OnActivity(m.MyActivity, mock.Anything).Return(result, nil)
```
all executors defined in the test options
```go
s.env.SetWorkerOptions(
  worker.Options{
	  Interceptors: ....,
  },
)
```
are executed.

## Actual Behavior
If I mock nonlocal activity with return values, interceptors are not executed. But for example if I mock the local activity in the tests with defined return values, interceptors are executed as expected. Also if I mock activities (local, nonlocal) with a function
```go
s.env.OnActivity(m.MyActivity, mock.Anything).Return(func (ctx context.Context) (int, error) {
    return result, nil
})
```
it works as expected and interceptors are executed.

## Steps to Reproduce the Problem

  1. Define interceptors in the test environment
  1. Mock non local activity with return values
  1. Run workflow test

## Specifications

  - Version: 1.26
  - Platform: I think problem is on all platforms

## Thoughts
I think the problematic code lives in the `internal/internal_workflow_testsuite.go`
https://github.com/temporalio/sdk-go/blob/c69831e92d7f52e564421fcb99ed49f693c5acbe/internal/internal_workflow_testsuite.go#L1963-L1975
If the mock is a function, we execute the activity executor that calls all interceptors internally. On other if we directly specify return values, we don't use the activity executor and `m.getMockValue(mockRet)` is called. 

There is a workaround to provide a mock function, but it is not comfortable and readable if I have to define a function that just returns values on all places. So theoretically an easy fix could be to just wrap the mocked return values to a function at sdk level and unify the execution paths.

**Comment by Quinn-With-Two-Ns** [comment]
Yes we should be consistent between local and normal activities. I'll have to verify what the Java SDK so we can align across SDKs

**Comment by PatrikValo** [comment]
Thanks! Meanwhile I have been looking into this issue and I created a commit - https://github.com/PatrikValo/sdk-go/commit/56f2c792a58e50c5568ecfba3f1fb86e0cce1702, I will not open PR for now. Please let me know what you find about the java sdk.
